### PR TITLE
Copy to Host: Array4.to_numpy(copy=True)

### PR DESCRIPTION
**Pull request by ax3l**
Support `to_numpy` copies w/o shared memory from device memory views in `Array4`.

X-ref #55

### CODE REVIEW

File: src/Base/Array4.H [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+                // allocate host memory copy
+                auto h_data = py::array_t<std::remove_cv_t<T>>(shape, strides);
+
+                // sync copy: host data is unpinned
+                Gpu::copy(Gpu::deviceToHost,
+                    a4.dataPtr(), a4.dataPtr() + a4.size(),
+                    h_data.mutable_data()
+                );
+                return h_data;
```
Comment (ax3l) [inline src/Base/Array4.H]: @WeiqunZhang does this look correct to you?

Comment (WeiqunZhang) [reply src/Base/Array4.H]: Yes

Comment (ax3l) [reply src/Base/Array4.H]: Thx. I am still debugging a copy issue that confuses me, the moment I have guard cells something goes off...

Comment (WeiqunZhang) [reply src/Base/Array4.H]: At the BaseFab/Array4 level, there is no concept of ghost cells in the BaseFab class.

Comment (ax3l) [reply src/Base/Array4.H]: Ah, my strides are still in bytes and need to be in elements for `py::array_t`.
Update: nope, needs to be bytes.

Comment (WeiqunZhang) [reply src/Base/Array4.H]: I see. But is shape in elements? If so, that seems inconsistent.

Comment (ax3l) [reply src/Base/Array4.H]: Oops, no... I forgot to set my desired value in all components of the multi fab

**Pull request merged**